app/controllers/application_controller.rb

```diff
@@ -27,7 +27,7 @@ def set_locale
     cookies.permanent[:locale] = I18n.locale if cookies[:locale].nil?
     logger.debug "* Locale set to '#{I18n.locale}'"
 
-    I18n.locale = portal_lang if portal_language_enabled?(I18n.locale)
+    I18n.locale = portal_lang unless portal_language_enabled?(I18n.locale)
```

Comment:
How do you explain this change?

Reply:
we set `I18n.locale` to `portal_lang` which is in our case `en` only when the `I18n.locale` is not enabled, when it's enabled we do nothing we keep the previous behavior (change the language of the portal).

So now if you do manually:
`https://stageportal.lirmm.fr/locale/de` for example then `I18n.locale` is not enabled, we use `portal_lang` which is `en` so the portal keep using english

but changing to french for example, the portal language is enabled so we do nothing and we change the portal language to french.

It's deployed to stage, you can test it.